### ISSUE TITLE
<FormGroup> renders as <div class="mb-3">

### ISSUE THREAD

**Issue by joshuajameshunt** [reactstrap/reactstrap#2431]
- components: `FormGroup`
- reactstrap version `9.0.1`
- import method `scss`
- react version `17.0.2`
- bootstrap version `5.1.3`

### What is happening?

<FormGroup> is rendering as <div class="mb-3">

### What should be happening?

<FormGroup> should be rendering as <div class="form-group">

### Steps to reproduce issue

1. Add a <FormGroup> component
2. Look at the render output

### Error message in console

```
N/A
```

### Code

<FormGroup></FormGroup>

**Comment by scivarolo** [comment]
According to [Bootstrap 5's migration docs](https://getbootstrap.com/docs/5.1/migration/#forms), form-specific layout classes (`.form-group`, `.form-row`, `.form-inline`) were dropped in favor of using the grid system

**Comment by joshuajameshunt** [comment]
@scivarolo you are correct, it just seems like adding `.form-group { margin-bottom: 1rem; }` to your global CSS is a lot more convenient than having to remove every single non-check FormGroup from your project (or replace all references with your own FormGroup component) because you styled it differently (a bottom margin of only 0.75rem, for example).

**Comment by juandelacruz23** [comment]
It's also a bad idea for me that the `mb-3` is added after classnames, making it impossible to remove that margin with utility classes nor with a personalized bootstrap variables file.

**Comment by davidacevedo** [comment]
`form-group` won't be added back in as it doesn't exist anymore in the bootstrap context, but what about having a prop that disables the default margin class?

**Comment by shtbik** [comment]
In this case, wouldn't be a bad idea to update the docs:
https://reactstrap.github.io/?path=/docs/components-forms--input#form-inline

**Comment by AJax2012** [comment]
> According to [Bootstrap 5's migration docs](https://getbootstrap.com/docs/5.1/migration/#forms), form-specific layout classes (`.form-group`, `.form-row`, `.form-inline`) were dropped in favor of using the grid system

Is this also why the `inline` property doesn't seem to work in the LabelProps or the FormProperties interfaces anymore as well? If so, would you mind if I submit a PR to remove that property from both? I don't know if your team allows outside parties to submit PRs, so I figured I'd ask, but as @shtbik pointed out, it doesn't see to work in the storybook demo either.

**Comment by illiteratewriter** [comment]
@AJax2012 Feel free to make a PR.

**Comment by AJax2012** [comment]
I unfortunately do not have the time to update the JavaScript files for the form-inline class (as well as fix everything that would break if you removed it), so I left the form types definition alone. Eventually, someone should probably get around to that though, haha.

I did remove the Label types definition...According to the git history, this definition never should have existed - the property in Label.js was removed in 2017 and the type definitions seem to have been created in 2020, so... Not sure what happened there. 

I also updated the broken storybook story in the documentation. It's technically unrelated, but it really needs to be fixed... It's updated to match the bootstrap documentation (I added a checkbox and removed the labels).

Here's the PR: https://github.com/reactstrap/reactstrap/pull/2568

**Comment by AJax2012** [comment]
Inline/form-inline has been removed following the above merged PR. Does that resolve this issue? I'm not sure if there was any other requested updates in this issue, but that seemed like the only remaining update request I found.

**Comment by illiteratewriter** [comment]
@AJax2012 FormGroup adds `mb-3` also to the div. I think it would be better if it doesn't add that automatically.

**Comment by AJax2012** [comment]
I think I should be able to take a look into that tonight... If it doesn't look too bad, I'll just go ahead and knock that out too.

**Comment by AJax2012** [comment]
I created a PR to resolve the `mb-3` issue: #2597

**Comment by AJax2012** [comment]
@illiteratewriter so I accidentally deleted my branch, but now I have a few questions as per this discussion. Last time this thread was updated, we agreed that not adding `mb-3` by default was preferable, but according to a comment on the PR, we wanted to change that to add an attribute (`noMargin `) that removed `mb-3` rather than removing it by default... So what is the preference here?

Also, I just noticed a few other things while looking at this... Do we want to automatically remove `mb-3` for the attribute `check` and `switch`? I noticed that was being tested, but I'm not sure what we want to do about this at this point...

**Comment by illiteratewriter** [comment]
The issue with removing default margin is that it is a breaking change. We decided to go with `noMargin` so as to not break anything. However, we can release as you did in the PR in the next version (v10). We'll have a discussion internally and get back on this.

**Comment by AJax2012** [comment]
Sounds good. I'll have time next week to redo all of this - should have something for you then. I'll keep an eye on this.

**Comment by AJax2012** [comment]
@illiteratewriter and @davidacevedo, I think this issue should finally be resolved, though it sounds like there might be more need for discussion in the future. Please see #2676.

**Comment by uguremirmustafa** [comment]
It is always a good idea to have an interface layer between UI libraries. I import the FormGroup component from MyFormGroup so that I don't need to make a change in bunch of places.
